### ISSUE TITLE
make integrations self-contained

### ISSUE THREAD

**Issue by jrhizor** [airbytehq/airbyte#173]
- integrations need to be responsible for defining the config spec
- integration images need an interface to produce the config spec 
- integrations should fail if the provided config is invalid
- integration tests should fail if validation is not being performed properly or if it can't access the spec
- integrations should be able to optionally convert their config to a different format (currently only required for BQ, but could be used in the future for postgres, others)

**Comment by sherifnada** [comment]
solved after #528

**Comment by cgardens** [comment]
@sherifnada - is moving standard source / standard destination into the integration not part of this task? should we have a separate issue for it?

**Comment by sherifnada** [comment]
@cgardens just saw this comment. Why would moving the entire source/destination definition into the image be part of the task? We still need a registry in the core application of sources/destinations and the docker images/tags they use. This is what standard source/destination currently refer to.

**Comment by cgardens** [comment]
Let's discuss this tomorrow when we're in person.
